### ISSUE TITLE
"speed" defaults to 0, not 3

### ISSUE THREAD

**Issue by ascorbic** [kornelski/pngquant#313]
Contrary to the documentation, the speed does not default to the value "3". It seems that it defaults to 0.

```sh
➜  pngquant  --ext .default.png test.png
➜  pngquant --speed 0 --ext .0.png test.png
➜  pngquant --speed 3 --ext .3.png test.png
➜  md5 test*.png
MD5 (test.0.png) = 627d092738c4a39d3b035f2007ba94c0
MD5 (test.3.png) = 625694e90263cdd13656f3acfa874f35
MD5 (test.default.png) = 627d092738c4a39d3b035f2007ba94c0
MD5 (test.png) = 734c89763c53859b3e001db60e6ea008

➜  ls -lS test*.png
-rw-rw-rw-@ 1 matt  staff  17194 30 Oct 15:21 test.png
-rw-r--r--  1 matt  staff  10468 30 Oct 15:21 test.3.png
-rw-r--r--  1 matt  staff  10426 30 Oct 15:21 test.0.png
-rw-r--r--  1 matt  staff  10426 30 Oct 15:21 test.default.png
```

The default, and --speed 0 produce an identical file, which is also the smallest size.

**Comment by kornelski** [comment]
--speed 0 is equivalent to the default, since the code is like `if (speed) set_speed(speed)`

**Comment by ascorbic** [comment]
The docs say that the default is 3 though. Those examples show that it's not, and is actually more compressed than that. What is the value that it's using?

**Comment by ubone** [comment]
there is no 0 in the docs - the values go from 1 to 11 (previously 10)
```
$ pngquant --ext -default.png Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage.png
$ pngquant -s 0 --ext -0.png Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage.png
$ pngquant -s 1 --ext -1.png Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage.png
$ pngquant -s 3 --ext -3.png Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage.png
$ pngquant -s 8 --ext -8.png Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage.png
$ /bin/du Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage*
3368    Screenshot_2018-11-12 Google - Wikipedia-fullpage-0.png
3320    Screenshot_2018-11-12 Google - Wikipedia-fullpage-1.png
3336    Screenshot_2018-11-12 Google - Wikipedia-fullpage-3.png
4580    Screenshot_2018-11-12 Google - Wikipedia-fullpage-8.png
3368    Screenshot_2018-11-12 Google - Wikipedia-fullpage-default.png
6108    Screenshot_2018-11-12 Google - Wikipedia-fullpage.png
$ /bin/md5sum Screenshot_2018-11-12\ Google\ -\ Wikipedia-fullpage*
b661a5938608e977855159764c09cbb4  Screenshot_2018-11-12 Google - Wikipedia-fullpage-0.png
665c7baaac911ea05e18990c09dbae30  Screenshot_2018-11-12 Google - Wikipedia-fullpage-1.png
4547455a2441f0bed3c485f48c00c2aa  Screenshot_2018-11-12 Google - Wikipedia-fullpage-3.png
f11aa39b63f34fd61c9eb65163f5cdaf  Screenshot_2018-11-12 Google - Wikipedia-fullpage-8.png
b661a5938608e977855159764c09cbb4  Screenshot_2018-11-12 Google - Wikipedia-fullpage-default.png
6f178400d31ea3c6eb285ea3a274e9ad  Screenshot_2018-11-12 Google - Wikipedia-fullpage.png
$ pngquant -V
2.12.2 (January 2018)
```

**Comment by kornelski** [comment]
speed 0 is an error. The bug here is that the CLI doesn't say "speed 0 doesn't make sense" instead of quietly ignoring the invalid value.

**Comment by ascorbic** [comment]
It makes sense insasmuch as it is clearly compressing it more than speed = 1. It's not just when `--speed 0` is specified though: it's also if the speed flag is omitted. It might be fair to say that *most* users are getting the "invalid" speed of 0.
The "correct" behaviour should be to set speed to 3 if `--speed` is 0 or missing. Unfortunately this would be a breaking change, as everyone would then get a different result from the previous one. I encountered this error working with [gatsby-plugin-sharp](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-sharp), which uses [imagemin-pngquant](https://github.com/imagemin/imagemin-pngquant) under the hood. Manually setting `--speed 3` broke all of the snapshot tests.

**Comment by kornelski** [comment]
Oh, sorry. I forgot I've changed the default to 4 in the underlying library, and forgot to update the CLI to match. This is now fixed.

BTW, pngquant doesn't give any guarantees about bit-exact reproducibility of results. Some processes use fast-math floating-point which may be optimized differently by compilers as they wish, and dithering relies on random numbers, which can be generated by libc any way it wants.
